libraries/AP_GPS/AP_GPS_SBF.cpp

```diff
@@ -62,10 +62,16 @@ AP_GPS_SBF::read(void)
         const char *init_str = _initialisation_blob[_init_blob_index];
 
         if (now > _init_blob_time) {
-            Debug("SBF sending init blob: %s\n", init_str);
-            port->write((const uint8_t*)init_str, strlen(init_str));
-            // if this is too low a race condition on start occurs and the GPS isn't detected
-            _init_blob_time = now + 2000;
+            if (now > _init_blob_advance_time + 2500) {
+                // try to enable input on the GPS port if we have not made progress on configuring it
+                Debug("SBF Sending port enable");
+                port->write((const uint8_t*)_port_enable, strlen(_port_enable));
+                _init_blob_advance_time = AP_HAL::millis();
+            } else {
+                Debug("SBF sending init string: %s", init_str);
+                port->write((const uint8_t*)init_str, strlen(init_str));
+            }
+            _init_blob_time = now + 1000;
```

Comment:
Shouldn't this be inside the `else`? Or you purposely want to wait more time after sending a *port enable*?

Reply:
I want to wait more time, as I want to give the GPS a chance to be in the right mode, and I don't want to stuff up the outgoing buffer. The buffer filling problem was actually associated with sending useless init blobs which this same commit fixed, so I could probably do it initially, but the general thought with GPS drivers at the moment is to not hit the GPS with tons of configuration at once (to avoid any stuttering/missed outputs).

It's worth noting that with the command ack code it is possible to send commands as soon as the previous one was ack'ed and the GPS appears to be stable with this. (I haven't investigated the exact timing, but nothing was noticable as an error). But as a rule of thumb we just want to always space commands out.

Reply:
> It's worth noting that with the command ack code it is possible to send commands as soon as the previous one was ack'ed and the GPS appears to be stable with this.

Possible but not done, right? It will only send the next command 1 second after the last was sent.

Reply:
> Possible but not done, right? It will only send the next command 1 second after the last was sent.

Correct, I initially had it moving on instantly and then reconsidered it. Sorry, was just noting that it does work without any delays, not that I was actually trying to ship that version to anyone.